### PR TITLE
feat: add prague check in ExecutionPayloadValidator

### LINKED ISSUE #9995: Add prague checks for ExecutionPayloadValidator
### Describe the feature

https://github.com/paradigmxyz/reth/blob/30840b7b4acd5d68439cafe8a3d65c7e2b3f626b/crates/payload/validator/src/lib.rs#L19-L19

these include checks for no 7702 transactions similar to 

https://github.com/paradigmxyz/reth/blob/30840b7b4acd5d68439cafe8a3d65c7e2b3f626b/crates/payload/validator/src/lib.rs#L138-L141

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by mvares**
Closes #9995

### CODE REVIEW

File: crates/payload/validator/src/lib.rs [hunk 1]
```diff
@@ -159,6 +159,12 @@ impl ExecutionPayloadValidator {
             }
         }
 
+        if self._is_prague_active_at_timestamp(sealed_block.timestamp) &&
```
Comment (Rjected) [inline crates/payload/validator/src/lib.rs:162]: do you mind renaming this method as well to not have the underscore?

**Review by Rjected — CHANGES_REQUESTED** [review]
one comment otherwise looks good!

File: crates/payload/validator/src/lib.rs (hunk 1 above)
Comment (mvares) [reply crates/payload/validator/src/lib.rs:162]: Good catch!

**Review by Rjected — APPROVED** [review]
awesome ty

**Comment by mvares** [comment]
:shipit:

**Review by mattsse — CHANGES_REQUESTED** [review]
pedantic nit

**Review by mattsse — APPROVED** [review]

**Comment by mvares** [comment]
the tests is falling... @mattsse

**Pull request merged**